### ISSUE TITLE
Make cmake.configure return string

### ISSUE THREAD

**Issue by MartinsSmirnovs** [microsoft/vscode-cmake-tools#2705]
### Brief Issue Summary

I have a project where I automatically regenerate `cmake-kits.json` each time I open Visual Studio Code. The project also uses CMake Variants, which utilizes `cmake-kits.json`, and it's configuration also gets invoked each time Visual Studio Code gets opened because of set `cmake.configureOnOpen`. It is important to maintain execution order for `cmake-kits.json` to be utilized by CMake Variants, and it can be easily achieved using `tasks.json`:
```
  "tasks": [
    ...
    {
      "label": "Configure CMake",
      "type": "process",
      "command": "${cmake.configure}",
      "dependsOn": ["My Custom Generate cmake-kits.json Task"]
    }
    ...
  ]
```
As can be seen, all we have to do is just call `cmake.configure` command. It also works as expected and the solution is almost ideal, however, it generates Visual Studio Code error `Cannot substitute command variable 'cmake.configure' because command did not return a result of type string.`. It happens, because `cmake.configure` is not a command [that can be substituted]( `cmake.configure` ).
In my case I would like `cmake.configure` to return at least some kind of string. Even empty string would be good if it can suppress Visual Studio Code error.

### Additional Information

Issue #1234 is related, but does not solve my problem.

**Comment by bobbrow** [comment]
The `cmake.configure` command does not return the command string and I don't believe VS Code has a generic task that will run a command for you.

Instead, in version 1.11 we added a new task type that should accomplish what you want. #2452  Can you try this?

```json
    {
      "label": "Configure CMake",
      "type": "cmake",
      "command": "configure",
      "dependsOn": ["My Custom Generate cmake-kits.json Task"]
    }
```

**Comment by MartinsSmirnovs** [comment]
@bobbrow 
Thanks, that is exactly what I needed. However, there is a problem. I have created following task:
```
        {
            "label": "CMake Configure",
            "type": "cmake",
            "command": "configure",
        }
```

If I execute it as a task using `Ctrl+P` + `task CMake Configure`, it runs in `OUTPUT` section as expected, but later shows:
```
Config Started...

 *  The terminal process terminated with exit code: -1. 
 *  Terminal will be reused by tasks, press any key to close it. 
```
in `TERMINAL` section, after execution.

However, when I run CMake configure using  `Ctrl+P` + `>CMake: Configure`, it just runs in `OUTPUT` section, but does not output any errors in `TERMINAL` section.

**Comment by bobbrow** [comment]
@elahehrashedi Does the configure task just try to run the command in 1.11 and fail out in the terminal?

@MartinsSmirnovs Sorry about that. We are addressing #2693 right now and will be running the configure in the terminal as well for version 1.12 of the extension (coming soon).

**Comment by MartinsSmirnovs** [comment]
Closing this issue since @bobbrow has provided the solution, and it's bug was fixed in [v1.12.26](https://github.com/microsoft/vscode-cmake-tools/releases/tag/v1.12.26).
